planner/core/logical_plan_builder.go

```diff
@@ -2477,7 +2477,7 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 	}
 	b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SelectPriv, dbName.L, tableInfo.Name.L, "", authErr)
 
-	if tbl.Type() == table.VirtualTable {
+	if tbl.Type() == table.VirtualTable && !infoschema.IsClusterMemTable(dbName.L, tableInfo.Name.L) {
```

Comment:
I think we should add a flag to `tbl.Type()` and make this condition more efficiency. (this is a hot pat)

Reply:
Mostly, `tbl.Type() == table.VirtualTable` will be false, then the later condition will not be evaluated.

Reply:
See: https://github.com/pingcap/tidb/pull/13065#discussion_r351136878